platform/nuclei/ux600/platform.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright (c) Nuclei Corporation or its affiliates.
+ *
+ * Authors:
+ *   lujun <lujun@nucleisys.com>
+ */
+
+#include <libfdt.h>
+#include <sbi/riscv_asm.h>
+#include <sbi/riscv_io.h>
+#include <sbi/riscv_encoding.h>
+#include <sbi/sbi_console.h>
+#include <sbi/sbi_const.h>
+#include <sbi/sbi_platform.h>
+#include <sbi_utils/fdt/fdt_fixup.h>
+#include <sbi_utils/irqchip/plic.h>
+#include <sbi_utils/serial/sifive-uart.h>
+#include <sbi_utils/sys/clint.h>
+
+/* clang-format off */
+
+#define UX600_HART_COUNT		1
+
+#define UX600_SYS_CLK			1000000000
+
+/* Nuclei timer base address */
+#define UX600_NUCLEI_TIMER_ADDR		0x2000000
+/* The clint compatiable timer offset is 0x1000 against nuclei timer */
+#define UX600_CLINT_TIMER_ADDR		(UX600_NUCLEI_TIMER_ADDR + 0x1000)
+
+#define UX600_PLIC_ADDR			0x8000000
+#define UX600_PLIC_NUM_SOURCES		0x35
+#define UX600_PLIC_NUM_PRIORITIES	7
+
+#define UX600_UART0_ADDR		0x10013000
+#define UX600_UART1_ADDR		0x10023000
+
+#define UX600_DEBUG_UART		UX600_UART0_ADDR
+#define UX600_UART_BAUDRATE		115200
+
+/* Full tlb flush always */
+#define UX600_TLB_RANGE_FLUSH_LIMIT	0
```

Comment:
I just wanted to check if it is intentional. Otherwise, default (4KB) actually performs better if the hardware supports it. We set it to zero in HiFive Unleashed because of some hardware errata.